superset-frontend/src/dashboard/components/Header/index.jsx

```diff
@@ -668,6 +673,7 @@ const Header = () => {
       NavExtension,
       boundActionCreators.onRedo,
       boundActionCreators.onUndo,
+      boundActionCreators.clearDashboardHistory(),
```

Comment:
You're executing the function in the dependent list 
```suggestion
      boundActionCreators.clearDashboardHistory,
```